src/net/sourceforge/kolmafia/session/BanishManager.java

```diff
@@ -13,27 +18,102 @@
+
+    private final boolean rolloverReset;
+
+    Reset() {
+      this(false);
+    }
+
+    Reset(final boolean rolloverReset) {
+      this.rolloverReset = rolloverReset;
+    }
+
+    final boolean isRolloverReset() {
+      return rolloverReset;
+    }
   }
 
-  private static class Banisher {
+  public enum Banisher {
+    BALEFUL_HOWL("baleful howl", -1, 1, true, Reset.ROLLOVER_RESET),
+    BANISHING_SHOUT("banishing shout", -1, 3, false, Reset.AVATAR_RESET),
+    BATTER_UP("batter up!", -1, 1, false, Reset.ROLLOVER_RESET),
+    BEANCANNON("beancannon", -1, 5, false, Reset.ROLLOVER_RESET),
+    BE_A_MIND_MASTER("Be a Mind Master", 80, 1, true, Reset.TURN_RESET),
+    BLART_SPRAY_WIDE("B. L. A. R. T. Spray (wide)", -1, 1, true, Reset.ROLLOVER_RESET),
+    BOWL_A_CURVEBALL("Bowl a Curveball", -1, 1, true, Reset.COSMIC_BOWLING_BALL_RESET),
+    BREATHE_OUT("breathe out", 20, 1, true, Reset.TURN_ROLLOVER_RESET),
+    BUNDLE_OF_FRAGRANT_HERBS(
+        "bundle of &quot;fragrant&quot; herbs", -1, 3, true, Reset.ROLLOVER_RESET),
+    CHATTERBOXING("chatterboxing", 20, 1, true, Reset.TURN_ROLLOVER_RESET),
+    CLASSY_MONKEY("classy monkey", 20, 1, false, Reset.TURN_RESET),
+    COCKTAIL_NAPKIN("cocktail napkin", 20, 1, true, Reset.TURN_RESET),
+    CRYSTAL_SKULL("crystal skull", 20, 1, false, Reset.TURN_RESET),
+    CURSE_OF_VACATION("curse of vacation", -1, 1, false, Reset.ROLLOVER_RESET),
+    DEATHCHUCKS("deathchucks", -1, 1, true, Reset.ROLLOVER_RESET),
+    DIRTY_STINKBOMB("dirty stinkbomb", -1, 1, true, Reset.ROLLOVER_RESET),
+    DIVINE_CHAMPAGNE_POPPER("divine champagne popper", 5, 1, true, Reset.TURN_RESET),
+    FEEL_HATRED("Feel Hatred", 50, 1, true, Reset.TURN_ROLLOVER_RESET),
+    GINGERBREAD_RESTRAINING_ORDER(
+        "gingerbread restraining order", -1, 1, false, Reset.ROLLOVER_RESET),
+    HAROLDS_BELL("harold's bell", 20, 1, false, Reset.TURN_RESET),
+    HOWL_OF_THE_ALPHA("howl of the alpha", -1, 3, false, Reset.AVATAR_RESET),
+    HUMAN_MUSK("human musk", -1, 1, true, Reset.ROLLOVER_RESET),
+    ICE_HOTEL_BELL("ice hotel bell", -1, 1, true, Reset.ROLLOVER_RESET),
+    ICE_HOUSE("ice house", -1, 1, false, Reset.NEVER_RESET),
+    KGB_TRANQUILIZER_DART("KGB tranquilizer dart", 20, 1, true, Reset.TURN_ROLLOVER_RESET),
+    LICORICE_ROPE("licorice rope", -1, 1, false, Reset.ROLLOVER_RESET),
+    LOUDER_THAN_BOMB("louder than bomb", 20, 1, true, Reset.TURN_ROLLOVER_RESET),
+    MAFIA_MIDDLEFINGER_RING("mafia middle finger ring", 60, 1, true, Reset.TURN_ROLLOVER_RESET),
+    NANORHINO("nanorhino", -1, 1, false, Reset.ROLLOVER_RESET),
+    PANTSGIVING("pantsgiving", 30, 1, false, Reset.TURN_ROLLOVER_RESET),
+    PEEL_OUT("peel out", -1, 1, true, Reset.AVATAR_RESET),
+    PULLED_INDIGO_TAFFY("pulled indigo taffy", 40, 1, true, Reset.TURN_RESET),
+    REFLEX_HAMMER("Reflex Hammer", 30, 1, true, Reset.TURN_ROLLOVER_RESET),
+    SABER_FORCE("Saber Force", 30, 1, true, Reset.TURN_ROLLOVER_RESET),
+    SHOW_YOUR_FAMILIAR_BORING_PICTURES(
```

Comment:
why are you trying to bore your familiar ;)

perhaps `SHOW_YOUR_BORING_FAMILIAR_PICTURES`